website.py

```diff
@@ -222,6 +245,24 @@
 updateDarkMode();
 darkModeCheckbox.addEventListener("change", updateDarkMode);
 
+function updateFunctions() {
+    var functionSelect = document.getElementsByName("functionselect")[0];
+    var prevValue = functionSelect.value;
+    var options = "<option value='all'>all functions</option>";
+    for (var line of sourceEl.value.split("\\n")) {
```

Comment:
"\n", surely? or oh, is this because of double quoting from Python? use an `r"""` string in that case.

Instead of splitting by newline you can do `for (let match of sourceEl.value.matchAll(/^\w*glabel\w+([A-Za-z0-9_]+)/mg))` I believe